tools/ramdump/ramdump_tool.c

```diff
@@ -328,11 +330,48 @@ static int ramdump_recv(int dev_fd)
 				}
 			}
 			printf("]\n");
+
+			ret = read(dev_fd, &buf, 1);
+			if (ret != 1) {
+				printf("Receiving ramdump %dTH byte failed, ret = %d\n", count, ret);
+				fclose(bin_fp);
+				return -1;
+			}
+			ret = fwrite(&buf, 1, 1, bin_fp);
+			if (ret != 1) {
+				printf("Writing ramdump %dTH byte failed, ret = %d\n", count, ret);
+				fclose(bin_fp);
+				return -1;
+			}
+			count++;
+
+			config_info_cnt = (uint8_t)buf;
+			config_info_size = sizeof(uint32_t);
+			while (config_info_cnt) {
+				while (config_info_size) {
+					ret = read(dev_fd, &buf, 1);
```

Comment:
Is there any specific reason why it ready and write byte by byte?
Why don't you read & write just once with bytes of config_info_size?
I/O always takes most long time than any other operation, as possible as we can, we should reduce I/O